simtools/data_model/metadata_collector.py

```diff
@@ -34,33 +45,118 @@ def __init__(self, args_dict):
-            self.top_level_meta["cta"]["context"]["associated_elements"]
-        )
+        self._fill_contact_meta(self.top_level_meta["cta"]["contact"])
         self._fill_product_meta(self.top_level_meta["cta"]["product"])
-        self._fill_top_level_meta_from_file(self.top_level_meta["cta"])
-        self._fill_association_id(self.top_level_meta["cta"]["context"]["associated_elements"])
         self._fill_activity_meta(self.top_level_meta["cta"]["activity"])
+        self._fill_process_meta(self.top_level_meta["cta"]["process"])
+        self._fill_context_from_input_meta(self.top_level_meta["cta"]["context"])
+        self._fill_associated_elements_from_args(
+            self.top_level_meta["cta"]["context"]["associated_elements"]
+        )
+
+    def get_data_model_schema_file_name(self):
+        """
+        Return data model schema file name.
+        The schema file name is taken (in this order) from the command line,
+        from the metadata file, from the data model name, or from the input
+        metadata file.
+
+        Returns
+        -------
+        str
+            Name of schema file.
+
+        """
+
+        try:
+            if self.args_dict["schema"]:
+                self._logger.debug(f"Schema file from command line: {self.args_dict['schema']}")
+                return self.args_dict["schema"]
+        except KeyError:
+            pass
+
+        try:
+            if self.top_level_meta["cta"]["product"]["data"]["model"]["url"]:
+                self._logger.debug(
+                    "Schema file from product metadata: "
+                    f"{self.top_level_meta['cta']['product']['data']['model']['url']}"
+                )
+                return self.top_level_meta["cta"]["product"]["data"]["model"]["url"]
+        except KeyError:
+            pass
+
+        if self.data_model_name:
+            self._logger.debug(f"Schema file from data model name: {self.data_model_name}")
+            return simtools.constants.SCHEMA_URL + self.data_model_name + ".schema.yml"
 
-    def _fill_association_meta_from_args(self, association_dict):
+        try:
+            self._logger.debug(
+                "Schema file from input metadata: "
+                f"{self.input_meta['cta']['product']['data']['model']['url']}"
+            )
+            return self.input_meta["cta"]["product"]["data"]["model"]["url"]
+        except KeyError:
+            pass
+
+        self._logger.warning("No schema file found.")
+
+    def get_data_model_schema_dict(self):
+        """
+        Return data model schema dictionary.
+
+        Returns
+        -------
+        dict
+            Data model schema dictionary.
+
+        """
+
+        try:
+            return gen.collect_data_from_yaml_or_dict(in_yaml=self.schema_file, in_dict=None)
+        except gen.InvalidConfigData:
+            self._logger.debug(f"Failed reading schema file from {self.schema_file}.")
+        return {}
```

Comment:
There are cases / applications where we do not have a metadata schema (in case is `self.schema_file == None`). This construction allows to keep consistent code at other places without a lot of 'if' statements checking of a file is there or not. This is also the reason why it is a debug message only - as there are cases where the 'raise' is expected behaviour.